src/Rules/Functions/ParameterCastableToStringFunctionRule.php

```diff
@@ -104,10 +114,36 @@ public function processNode(Node $node, Scope $scope): array
 			$argsToCheck = $origArgs;
 		} elseif (in_array($functionName, $checkFirstArgFunctions, true)) {
 			$normalizedArgs = $getNormalizedArgs();
-			if ($normalizedArgs === []) {
+			if (!array_key_exists(0, $normalizedArgs)) {
 				return [];
 			}
 			$argsToCheck = [0 => $normalizedArgs[0]];
+		} elseif (in_array($functionName, $checkSortWithFlagsFunctions, true)) {
+			$normalizedArgs = $getNormalizedArgs();
+			if (!array_key_exists(0, $normalizedArgs)) {
+				return [];
+			}
+
+			$argsToCheck = [0 => $normalizedArgs[0]];
+			$flags = null;
+			if (array_key_exists(1, $normalizedArgs)) {
+				$flags = $scope->getType($normalizedArgs[1]->value);
+			} elseif (array_key_exists(1, $functionParameters)) {
+				$flags = $functionParameters[1]->getDefaultValue();
+			}
+
+			if ($flags === null || $flags->equals(new ConstantIntegerType(SORT_REGULAR))) {
+				return [];
+			}
+
+			foreach (TypeUtils::getConstantIntegers($flags) as $flag) {
+				if ($flag->getValue() !== SORT_NUMERIC) {
```

Comment:
I'm not sure what's the best way to handle flags like this. My thinking is as follows:

- `SORT_REGULAR` does not have any constraints, so I only check for equality. If there is a union (or int), then the stricter check should be maintained.
- `SORT_NUMERIC` seems the strictest (though not exactly, because it does allow arrays, unlike `SORT_STRING`), so if it's a union `SORT_NUMERIC` and something else, I go with `SORT_NUMERIC`.
- Otherwise, I assume `SORT_STRING` (or `SORT_LOCALE_STRING` which is the same for our purposes).

The only edge-case that I can think of right now is e.g. `array_unique([[1], [2]], rand() ? SORT_NUMERIC : SORT_STRING)`. IDK if it's worth it to complicate it further with such edge cases. :sweat_smile: